packages/client/src/components/authForm/authForm.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ChangeEvent, PropsWithChildren, useEffect, useState } from 'react';
+import style from './authForm.module.scss';
+import classNames from 'classnames';
+import { Button, ButtonVariation } from '../button/button';
+import { Link } from 'react-router-dom';
+import { getInputName } from '../../utils/helpers';
+import validate from '../../service/validate/validate';
+
+export interface IAuthFormProps {
+  submitBtnName: string;
+  title: string;
+  isDarkTheme?: boolean;
+  linkTitle?: string;
+  linkName?: string;
+  linkPath?: string;
+  errorTitle?: string;
+  validation?: boolean;
+  linkAction: React.MouseEventHandler<HTMLAnchorElement>;
+  formAction: React.FormEventHandler<HTMLFormElement>;
+  formFocus?: React.FormEventHandler<HTMLFormElement>;
+}
+
+type InputsState = Record<string, { value: string; errorText?: string }>;
+
+export const AuthForm = (props: PropsWithChildren<IAuthFormProps>) => {
+  const {
+    isDarkTheme,
+    submitBtnName,
+    title,
+    formAction,
+    formFocus,
+    children,
+    linkPath,
+    linkTitle,
+    linkName,
+    errorTitle,
+    validation
+  } = props;
+  const [inputValues, setInputValues] = useState({} as InputsState);
+  const [error, setError] = useState<boolean>(false);
+  const { password, confirmPassword } = inputValues;
+
+  useEffect(() => {
+    let isValid = true;
+    Object.entries(inputValues).forEach(([key]) => {
+      if (inputValues[key].errorText) {
+        isValid = false;
+      }
+    });
+    if (isValid) {
+      setError(false);
+    } else {
+      setError(true);
+    }
```

Comment:
[Предложение]
Здесь записывается в `setError` отрицание `isValid`, то есть можно записать проще
```suggestion
    setError(!isValid);
```

Reply:
Да, точно